### PR TITLE
[Search] Conditionally pull catalog entities from local catalog if configured

### PR DESCRIPTION
**Pull request by iamEAP**
## Hey, I just made a Pull Request!

Aims to provide a reasonable interim solution to #5207 until there's consensus on #5210...

Not sure of what, if any, implications there are of building a completely separate instance of the catalog (but with identical configs).

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Not sure about this construction of the catalog builder. Would it make more sense to use the catalog-client?

**Comment by freben** [comment]
Since the plugin-to-plugin communications so far has only happened via their service interfaces, not via code sharing or direct injection

**Comment by iamEAP** [comment]
Hmm...  Just looked at the Catalog client and I see it's just making HTTP calls under the hood.  The problem we're trying to work around is when folks have auth enabled for backend APIs.  Was hoping there'd be a reasonable in-process way to interact with the catalog if it happened to be deployed together with the search backend.  Any other ideas @freben?

**Review by Rugvip — COMMENTED** [review]
This is a bit funky. I'd say we either leave it broken for users that have an authenticated catalog API until we can have a proper fix in place, or if we really want it to be working we could look at adding some kind of hacky global state in the background so that we can gain access to the actual entities catalog instance that is passed to the catalog plugin.

File: plugins/catalog-backend/src/search/DefaultCatalogCollator.ts [hunk 1]
```diff
@@ -18,24 +18,49 @@ import { PluginEndpointDiscovery } from '@backstage/backend-common';
...
+
+  static async fromEnv(env: PluginEnv) {
+    const { config, discovery } = env;
+
+    // If configs exist for the catalog, the catalog is deployed in the same
+    // process. Return a Collator configured to pull directly from the catalog.
+    if (config.get('catalog')) {
+      const builder = new CatalogBuilder(env);
+      const { entitiesCatalog } = await builder.build();
+      return new DefaultCatalogCollator({ entitiesCatalog });
```
[the remark this answers is not in the document]
Comment (iamEAP) [reply plugins/catalog-backend/src/search/DefaultCatalogCollator.ts:56]: Ah, indeed.

I'll close this for now, but hope to have a better fix ahead of closing out our Search V1 milestone.  👍  from me on figuring out a #5210 soon.

**Pull request closed without merging**